File: easy/3861_minimum_capacity_box/solution.py
from typing import List


class Solution:
    def minimumIndex(self, capacity: List[int], itemSize: int) -> int:
        min_i = 0

        for i in range(len(capacity)):
            if capacity[i] >= itemSize and (
                capacity[min_i] < itemSize or capacity[i] < capacity[min_i]
            ):
                min_i = i

        return min_i if capacity[min_i] >= itemSize else -1

File: easy/3861_minimum_capacity_box/solution.cpp
#include <vector>

class Solution {
public:
  int minimumIndex(std::vector<int> &capacity, int itemSize) {
    int minI = 0;

    for (int i = 0; i < capacity.size(); ++i) {
      if (capacity[i] >= itemSize &&
          (capacity[minI] < itemSize || capacity[i] < capacity[minI]))
        minI = i;
    }

    return capacity[minI] >= itemSize ? minI : -1;
  }
};
